scripts/errata_fix_script.py

```diff
@@ -0,0 +1,199 @@
+        for albs_package in package.albs_packages:
+            if albs_package.status != ErrataPackageStatus.released:
+                continue
+            pulp_pkg = await pulp.get_rpm_package(
+                albs_package.get_pulp_href(),
+                include_fields=pulp_pkg_fields,
+            )
+            pulp_pkg['reboot_suggested'] = (
+                albs_package.errata_package.reboot_suggested
+            )
+            location_href = re.sub(
+                r'^Packages/', '', pulp_pkg['location_href']
+            )
+            pulp_pkg['location_href'] = location_href
+            albs_packages_cache[location_href] = pulp_pkg
+    return albs_packages_cache
+
+
+async def update_pulp_db():
+    logging.info("Update pulp db is started")
+    pulp = pulp_client.PulpClient(
+        settings.pulp_host, settings.pulp_user, settings.pulp_password
+    )
+    albs_packages_cache = {}
+    latest_record_id = ''
+    logging.info('updating pulp db records')
+    with PulpSession() as pulp_db, SyncSession() as albs_db:
+        records: typing.List[UpdateRecord] = pulp_db.execute(
+            select(UpdateRecord)
+            .order_by(UpdateRecord.id)
+        ).scalars().all()
+        for record in records:
+            logging.info("Processing errata %s", record.id)
+            if latest_record_id != record.id:
+                albs_packages_cache = await prepare_albs_packages_cache(
+                    albs_db, pulp, record.id
+                )
+            if not albs_packages_cache:
+                logging.info('Skipping record %s, there is no ErrataRecord',
+                             record.id)
+                continue
+            collection = record.collections[0]
+            collection_arch = re.search(
+                r'i686|x86_64|aarch64|ppc64le|s390x',
+                collection.name,
+            ).group()
+            logging.info('Start checking errata packages for %s', record.id)
+            albs_packages = {
+                location_href
+                for location_href, pkg in albs_packages_cache.items()
+                if pkg['arch'] in (collection_arch, 'noarch')
+            }
+            errata_packages = {
+                pkg.filename
+                for pkg in collection.packages
+            }
+            missing_filenames = albs_packages.difference(errata_packages)
+            if not missing_filenames:
+                logging.info('Errata %s is ok', record.id)
+                continue
+            for filename in missing_filenames:
+                pulp_pkg = albs_packages_cache[filename]
+                if (
+                    pulp_pkg['sha256'] is None
+                    and pulp_pkg['checksum_type'] == 'sha256'
+                ):
+                    logging.warning(
+                        'sha256 sum is missing for %s using pkgId instead',
+                        filename,
+                    )
+                    pulp_pkg['sha256'] = pulp_pkg['pkgId']
```

Comment:
Maybe stupid question. Is this the right thing to do here? Why not just leaving it empty?

Reply:
this field have not null constraint in pulp DB